x/protocolpool/keeper/abci_test.go

```diff
@@ -0,0 +1,162 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"cosmossdk.io/collections"
+	"cosmossdk.io/core/header"
+	"cosmossdk.io/log"
+	"cosmossdk.io/math"
+	storetypes "cosmossdk.io/store/types"
+	"cosmossdk.io/x/auth"
+	authkeeper "cosmossdk.io/x/auth/keeper"
+	authtypes "cosmossdk.io/x/auth/types"
+	"cosmossdk.io/x/bank"
+	bankkeeper "cosmossdk.io/x/bank/keeper"
+	banktypes "cosmossdk.io/x/bank/types"
+	"cosmossdk.io/x/protocolpool"
+	"cosmossdk.io/x/protocolpool/keeper"
+	"cosmossdk.io/x/protocolpool/types"
+
+	addresscodec "github.com/cosmos/cosmos-sdk/codec/address"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	"github.com/cosmos/cosmos-sdk/testutil/integration"
+	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+)
+
+func TestContinuousFundEndBlocker(t *testing.T) {
```

Comment:
this test doesnt seem to test end block running many times without funding of the module account.

Reply:
this test includes 3 important testCase scenarios where funding is possible from pool module account(such that account is already funded here) with a 
 normal continuous fund proposal, another with recipient with more cap funds and another with an expired time. Hence covering all edge cases.

Reply:
this would then elude to the issue mentioned above. When funds dont come to the module account it should not take funds from the protocolpool and distribute. This could lead to the protocol pool hitting 0 over time if the epoch is long enough